pkg/controller/instance/instance_controller.go

```diff
@@ -316,7 +307,54 @@ func instanceEventFilter(mgr manager.Manager) predicate.Funcs {
 	}
 }
 
-func createPlan(mgr manager.Manager, planName string, instance *kudov1alpha1.Instance) error {
+func createPlan(r *ReconcileInstance, planName string, instance *kudov1alpha1.Instance) error {
+	ctx := context.TODO()
```

Comment:
IMO, it might be a good idea to break this out into two methods:

1. One that creates the Kubernetes structs given an instance.
2. Another that actually creates the objects in the API.

That would make method 1 easily unit testable.

Reply:
I've separated the retrieving of the k8s objects from the creating of new objects.  I'm not sure it buys anything.